types/events.go

```diff
@@ -87,8 +89,13 @@ func TypedEventToEvent(tev proto.Message) (Event, error) {
 		return Event{}, err
 	}
 
+	// sort the keys to ensure the order is always the same
```

Comment:
it might be safer and more obvious if we just moved the sorting to `EventManager#EmitEvents`. WDYT?

Reply:
In `EmitEvents` we don't use map. It's a deterministic append.

Reply:
Look at what `TypedEventToEvent` is doing. You can obviously sort the attributes in `EmitEvents` prior to appending.

Reply:
I know, I'm just saying that it's not needed in `EmitEvents`. Is there any benefit of sorting them in `EmitEvents`?

Reply:
It's future proof. It protects against any other new APIs or changes to existing ones, since `EmitEvents` is what actually returns the events via ABCI.

Reply:
lets this merged to protect current users then create an issue for future proofing.